harness/go/harness/feature.go

```diff
@@ -0,0 +1,155 @@
+package harness
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+	"sync"
+
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/worker"
+)
+
+// SDKVersion is the Go SDK version with the "v" prefix.
+const SDKVersion = "v" + temporal.SDKVersion
+
+// Feature represents a feature that can be executed.
+type Feature struct {
+
+	// Set of workflows to register. This can be a single workflow or a slice.
+	Workflows interface{}
+
+	// Set of activities to register. This can be a single activity or a slice.
+	Activities interface{}
+
+	// If present, expects workflow to fail with this activity error string.
+	ExpectActivityError string
```

Comment:
This seems over-specific. Why activity error? Why not just any error matching?

Reply:
Because I needed it at the time. I am expecting this to grow significantly and lazily as people need to assert different things. I just so happened to be the first user and wanted to set an expected activity error. Anyone can of course set their own `CheckResult` and check whatever, this is just a shortcut.